pwiz_tools/Skyline/Model/Proteome/ProteinAssociation.cs

```diff
@@ -805,24 +807,61 @@ public SrmDocument CreateDocTree(SrmDocument current, IProgressMonitor monitor)
+            var unmappedPeptideNodes = _peptideToPath.SelectMany(kvp => kvp.Value).Where(p => !p.IsDecoy).ToHashSet();
+            unmappedPeptideNodes.ExceptWith(ParsimoniousProteins.Values.SelectMany(pag => pag.Peptides));
 
             // Modifies and adds old groups that still contain unmatched peptides to newPeptideGroups
             foreach (var nodePepGroup in current.MoleculeGroups)
             {
-                // Get non-peptide children
-                var nonPeptideNodes = nodePepGroup.Children.Where(node => (node as PeptideDocNode)?.Peptide.Target.IsProteomic == false).ToList();
+                var peptideDocNodes = nodePepGroup.Children.Where(node => node is PeptideDocNode).Cast<PeptideDocNode>().ToList();
 
-                // Ignore old groups with no non-peptide children
-                if (nonPeptideNodes.Count == 0)
+                // drop all empty nodes
+                if (peptideDocNodes.Count == 0)
+                    continue;
+
+                // drop old Unmapped Peptides group
+                if (nodePepGroup.Name == Resources.ProteinAssociation_CreateDocTree_Unmapped_Peptides)
                     continue;
 
-                // Adds all pre-existing proteins to list of groups that will be added in the new document
-                /*if (nodePepGroup.PeptideGroup is FastaSequence) 
+                // Keep decoy and iRT lists
+                if (nodePepGroup.IsDecoy)
+                {
+                    appendPeptideLists.Add(nodePepGroup);
+                    continue;
+                }
+
+                if (peptideDocNodes.All(node => node.GlobalStandardType == StandardType.IRT))
                 {
                     newPeptideGroups.Add(nodePepGroup);
+                    unmappedPeptideNodes.ExceptWith(peptideDocNodes); // do not count iRT peptides as unmapped
+                    continue;
+                }
+
+                // Keep peptide lists that contain unmapped peptides
+                if (nodePepGroup.IsProteomic && nodePepGroup.IsPeptideList)
+                {
+                    // If a peptide list peptide is unmapped, leave it in the peptide list but remove it from the global unmapped list
+                    var peptidesByMappedStatus = peptideDocNodes.ToLookup(node => _peptideToProteins.Contains(kvp => kvp.Key.Target == node.Target), node => node);
+                    var unmappedPeptides = peptidesByMappedStatus[false].ToHashSet();
+                    unmappedPeptideNodes.ExceptWith(unmappedPeptides);
+
+                    // If it was mapped, remove it from the peptide list
+                    var mappedPeptides = peptidesByMappedStatus[true];
+                    var mappedPeptideIndexes = mappedPeptides.Select(node => node.Peptide.GlobalIndex);
+                    var newPeptideList = nodePepGroup.RemoveAll(mappedPeptideIndexes.ToList()) as PeptideGroupDocNode;
+                    appendPeptideLists.Add(newPeptideList);

```

Comment:
I think this should be:
```
if (newPeptideList.Children.Count != 0)
{
   appendPeptideLists.Add(newPeptideList);
}
```
If the peptide list used to have some peptides and after mapping it has zero peptides, it should be removed from the document.

Reply:
So remove a peptide list if it was populated before association and empty after association, but keep a peptide list if it was empty before association? That seems a bit inconsistent, what do you think @brendanx67 ?

Reply:
My comment was addresses both cases. My inclination is, to be consistent, to remove empty peptide lists regardless of whether they had peptides in them to begin with. Alternately to keep empty peptide lists regardless of whether they had peptides in them to begin with (users can always delete them afterward). But like Brendan I'm interested to hear if you have an example in mind of "the user had some reason to have it in the document in the first place".

Reply:
I would recommend a thorough unit test of all the interesting cases to ensure they are and stay what we want and expect.

Reply:
I side with deleting them always. I can't think of a use case where a user would want them. So, keeping them always is likely to annoy the user who has to delete them after this operation.

Reply:
The point of this feature was to take a potentially complex document full of lists, but with no protein sequence backings, and allow the user to specify the FASTA file containing the sequences, and have Skyline magically make everything as if they had used File > Import > FASTA from the start. Imagine a case where the user started with File > Import > Assay Library with no protein sequence information. They may end up with many thousands of peptide lists representing proteins. Now they want to "Associate Proteins" (sequence information) with their peptides. This is one of our most important use cases. Please don't lose sight of that in noodling over the details in documents with a few peptides in them.

Reply:
The rat plasma document is actually a reasonably good example. About half of its proteins have sequence information and the rest are peptide lists standing in for proteins. This feature should be able to leave the user with an ideal document when all they do is supply the rat FASTA.